gobblin-runtime/src/main/java/org/apache/gobblin/runtime/SafeDatasetCommit.java

```diff
@@ -326,11 +326,6 @@ private void finalizeDatasetStateBeforeCommit(JobState.DatasetState datasetState
         // The dataset state is set to FAILED if any task failed and COMMIT_ON_FULL_SUCCESS is used
         datasetState.setState(JobState.RunningState.FAILED);
         datasetState.incrementJobFailures();
-        Optional<String> taskStateException = taskState.getTaskFailureException();
-        String errMsg = "At least one task did not committed successfully. Setting dataset state to FAILED. "
-            + (taskStateException.isPresent() ? taskStateException.get() : "Exception not set.");
-        log.warn(errMsg);
```

Comment:
Maybe keep this warning, it looks helpful.

Reply:
I am quite sure this warning must have got in the original-place-of-failure. What do you think?

Reply:
probably. If so, it's logged in some mapper node, which is hard to locate, which might be the reason that `finalizeDatasetState1 do so.